### PR TITLE
BPS-638 Add service display name to service folder mappings config

### PR DESCRIPTION
**Pull request by aliveni**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-638

### Change description ###
Added service display name property to the existing service folder mappings.
Letters count report will use the new property instead of the folder name in the next PR.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/resources/application.yaml [hunk 1]
```diff
@@ -75,17 +75,22 @@ ftp:
...
+      displayName: Bulk Print
     - service: divorce_frontend
       folder: DIVORCE
+      displayName: Divorce
     - service: probate_backend
       folder: PROBATE
+      displayName: Probate
     - service: sscs
       folder: SSCS
+      displayName: SSCS
```
Comment (doncem) [inline src/main/resources/application.yaml]: should `displayName` be `display-name` instead?

File: src/main/java/uk/gov/hmcts/reform/sendletter/services/ftp/ServiceFolderMapping.java [hunk 2]
```diff
@@ -16,10 +19,15 @@ public ServiceFolderMapping(FtpConfigProperties configProperties) {
...
-        return configProperties.getServiceFolders().values();
+        return configProperties.getServicesConfig()
+            .stream()
+            .map(ServiceNameFolderMapping::getFolder).collect(toList());
     }
 
     public Optional<String> getFolderFor(String serviceName) {
-        return Optional.ofNullable(configProperties.getServiceFolders().get(serviceName));
+        return configProperties.getServicesConfig()
+            .stream().filter(config -> config.getDisplayName().equals(serviceName))
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/sendletter/services/ftp/ServiceFolderMapping.java]: should it not compare service name to service name?

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/sendletter/services/ftp/ServiceFolderMapping.java]: (it's used by the upload job, which uses service names, not display names

File: src/main/resources/application.yaml [hunk 3]
```diff
@@ -75,17 +75,22 @@ ftp:
   downtime:
     from: ${FTP_DOWNTIME_FROM:16:00}
     to: ${FTP_DOWNTIME_TO:17:00}
-  service-folders:
+  services-config:
```
Comment (rkondratowicz) [inline src/main/resources/application.yaml]: perhaps, for clarity, this config could be moved out of the `ftp:` tree in the yaml (`displayName` has nothing to do with ftp)

File: src/integrationTest/resources/application.properties [hunk 4]
```diff
@@ -19,6 +19,7 @@ ftp.downtime.to: 23:59
 
 ftp.service-folders[0].service=some_service_name
 ftp.service-folders[0].folder=some_folder
+ftp.service-folders[0].display-name=some folder
```
Comment (rkondratowicz) [inline src/integrationTest/resources/application.properties]: maybe 'some display name'?

File: src/main/resources/application.yaml (hunk 3 above)
Comment (aliveni) [reply src/main/resources/application.yaml]: @rkondratowicz I'll close this pr and create another pr to add config under `reports:`

**Comment by aliveni** [comment]
I'll create a new PR to move the reports service config from `ftp:` to `reports:`

**Pull request closed without merging**